### ISSUE TITLE
[SUGGESTION] Update Node: Can't write my extension because Bracket's node environment

### ISSUE THREAD

**Issue by TheOnlyArtz** [adobe/brackets#14049]
### Prerequisites

* [x] Can you reproduce the problem with `Debug -> Reload Without Extensions`?
* [x] Did you perform a cursory search to see if your bug or enhancement is already reported?
* [x] Did you read the [Troubleshooting guide]
### Description

I'm writing an extension which uses a node_module which need a Node 8.2+ to run properly 
And I found out that Brackets uses Node v6

### Steps to Reproduce

1. Update Brackets to use Node LTS latest

### Versions
Brackets v1.11
Win10 as OS

**Comment by thatcomputerguy0101** [comment]
Updating to Node 8 or 10 would be not only nice for newer programming features, but also for development. By default, Node 8+ uses the Chrome Debugging Protocol for debugging, so if the Chromium version of the shell is also updated to v55+, a debugger for the Node process could be opened along with enabling the debug port, resulting in an appearance similar to how the current Debug>Show Developer Tools operates, but for the node process instead. Another thing, it would be nice if the remote debugging port was then changed away from default in order to not conflict with other Node processes.

(The following paragraph is summarized in the list below)
Based on the brackets-shell repository, I believe that the primary portion of this fix would be to just change the version numbers for the CEF build and the Node build. Beyond that, I think that this should be just figuring out what features were depreciated/removed and updating Brackets not to use them. I have messed around with changing the Node executable out for a later version (v10), and I have not identified any problems that updating Node causes. I have no clue what problems might arise from updating CEF though, so I have no clue what problems may arise. Also, changing the port would require the Node process to be launched with the --inspect-port flag and a port number to be used instead, rather than just running the file with no arguments.

In summary, a To-Do(-in-this-order) list:
1. - [ ] Choose which versions of Node (8, 10, or soon 12) and what build of CEF (way too many, but anything with Chromium 55+ should work, the later, the better)
2. - [ ] Change Gruntfile.js in brackets-shell with the version numbers decided above
3. - [ ] Run tests with the updated build information and fix resulting bugs
4. - [ ] Change the remote debug port on the node process
5. - [ ] Add a function to create a Node DevTools window to the Debug>Enable Node Debugger command, and possibly change the name of the command
6. - [ ] Do whatever else you guys do before publishing a new version of Brackets
